docs/developer.md

```diff
@@ -0,0 +1,122 @@
+
+The command `npm run start-and-serve` runs in watch mode, so changes to files will be detected and bundled into `dev/` automatically, but you will need to refresh the page in your browser manually to see the changes -- we have disabled "hot reloading" because automatically refreshing the browser would cause any active notebooks to lose their evaluation state.
+
+If you require verbose Redux logging, you can use the command `npm run start-and-serve -- reduxVerbose`
+
+##### Exporting from client-only dev mode
+
+In this mode, resource paths are set to be relative to the `dev/` directory. Thus, if you export a bundled notebook from a dev notebook, you need to be sure save the exported HTML file in the `dev/` folder for the relative paths to correctly resolve the required js, css, and font files (and if you want to share a notebook that you created in a dev environment, you'll need to update the paths to point to the web-accessible resources at `iodide.io` and `iodide.app`).
+
+#### Server mode
+
+We have been building an experimental iodide server based on Python and Django. Currently the main features
+it supports are login/identity (via the github API). To test/run it locally, follow this set of steps:
+
+* Register a [github oauth token](https://github.com/settings/applications/new). Set the homepage URL to be
+"http://localhost:8000" and the authentication callback URL to be "http://localhost:8000/oauth/complete/github/".
+* Copy `.env-dist` to `.env` and set the `GITHUB_CLIENT_ID` and `GITHUB_CLIENT_SECRET` to the values provided above.
+* Make sure you have [Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/) installed and working correctly
+* Run `make build && make up`
+* You should now be able to navigate to a test server instance at http://localhost:8000
+
+On subsequent runs, you only need to run `make up`.
+
+Additionally, if you are working on client code, you can run `npm run start` in a separate terminal to run webpack in watch mode (which will make your client code changes visible on page reload). If you require verbose Redux logging, you can set the environment variable `REDUX_LOGGING=VERBOSE` with the command `REDUX_LOGGING=VERBOSE npm run start`
+
+Sometimes, for debugging purposes, it is useful to have a shell session inside the "app" docker container. You
+can use either the `make shell` command (creates a shell session with the "app" user) or the `make root-shell`
+commands (creates a shell session logged in as root, useful for experimenting with new python packages).
+
+### Building the docs
+
+The documentation is written in markdown, and uses
+[mkdocs](https://www.mkdocs.org/) to generate a static website.
+
+To test changes to the docs locally, you'll need to install `mkdocs` and
+`markdown-include`. This is completely independent of the server's Docker image
+described above. In most cases can be accomplished with:
```

Comment:
s/In most cases this can be accomplished with/If you have pip/python installed on your system, you can just install the packages directly:/